sources/frontend.c

```diff
@@ -474,27 +474,77 @@ static inline bool od_eject_conn_with_rate(od_client_t *client,
 	return res;
 }
 
-static inline bool od_should_drop_connection(od_client_t *client,
-					     od_server_t *server)
+static inline bool od_eject_conn_with_timeout(od_client_t *client,
+					      od_server_t *server,
+					      od_instance_t *instance,
+					      int timeout)
 {
-	od_instance_t *instance = client->global->instance;
+	const uint64_t interval_usec = 1000000;
 
-	//TODO:: drop no more than X connection per sec/min/whatever
-	if (od_likely(instance->shutdown_worker_id == INVALID_ID)) {
-		// try to optimize likely path
-		return false;
+	if (server == NULL) {
```

Comment:
Are we considering any client without server expired independently from timeout?